lib/virtual-fs/src/mem_fs/file.rs

```diff
@@ -1399,3 +1382,52 @@ impl ReadOnlyFile {
         Ok(())
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use std::ops::Deref;
+
+    use super::*;
+
+    #[test]
+    fn test_memfs_file() {
+        let mut file = File::new(None);
+
+        let mut cursor = 0;
+
+        file.write(b"hello", &mut cursor).unwrap();
+        assert_eq!(file.buffer.deref(), b"hello");
+        assert_eq!(cursor, 5);
+
+        file.write(b" ", &mut cursor).unwrap();
+        assert_eq!(file.buffer.deref(), b"hello ");
+        assert_eq!(cursor, 6);
+
+        file.write(b"world", &mut cursor).unwrap();
+        assert_eq!(file.buffer.deref(), b"hello world");
+        assert_eq!(cursor, 11);
+
+        cursor = 0;
+        file.write(b"alpha", &mut cursor).unwrap();
+        assert_eq!(file.buffer.deref(), b"alpha world");
+        assert_eq!(cursor, 5);
+
+        cursor = 2;
+        file.write(b"ter", &mut cursor).unwrap();
+        assert_eq!(file.buffer.deref(), b"alter world");
+        assert_eq!(cursor, 5);
+
+        cursor = 6;
+        file.write(b"means", &mut cursor).unwrap();
+        assert_eq!(file.buffer.deref(), b"alter means");
+        assert_eq!(cursor, 11);
+
+        cursor = 6;
+        file.write(b"the universe!", &mut cursor).unwrap();
+        assert_eq!(file.buffer.deref(), b"alter the universe!");
+        assert_eq!(cursor, 19);
+
+        file.write(b" mkay?", &mut cursor).unwrap();
+        assert_eq!(file.buffer.deref(), b"alter the universe! mkay?");
+    }
```

Comment:
maybe a test with a seek and truncate could be interesting?

Reply:
Definitely.

We should have a proper test suite for `trait FileSystem` implementations.

I only added a tests for `write()` because that's the only thing I changed.